docs: add default execution mode & agent pools to orgs and workspaces (#436)
hashicorp/terraform-docs-common: website/docs/cloud-docs/api-docs/workspaces.mdx

| `data.attributes.auto-apply`                  | boolean | `false`   | Whether to automatically apply changes when a Terraform plan is successful, [with some exceptions](/terraform/cloud-docs/workspaces/settings#auto-apply-and-manual-apply).                                                                                                                                                                                                                                                                                                                                      |
| `data.attributes.description`                 | string  | (nothing) | A description for the workspace.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                |

> | `data.attributes.execution-mode`              | string  | (nothing) | Which [execution mode](/terraform/cloud-docs/workspaces/settings#execution-mode) to use. Valid values are `remote`, `local`, and `agent`. When set to `local`, the workspace will be used for state storage only. This value must not be specified if `operations` is specified.                                                                                                                                                                                                                                |

Reviewer: ❓ should we say that it *must* be specified if `setting-overwrites.execution-mode` is set to `true`

| `data.attributes.vcs-repo`                    | object  | (nothing) | Settings for the workspace's VCS repository. If omitted, the workspace is created without a VCS repo. If included, you must specify at least the `oauth-token-id` and `identifier` keys below.                                                                                                                                                                                                                                                                                                                  |
| `data.attributes.working-directory`           | string  | (nothing) | A relative path that Terraform will execute within. This defaults to the root of your repository and is typically set to a subdirectory matching the environment when multiple environments exist within the same repository.                                                                                                                                                                                                                                                                                   |

> | `data.attributes.setting-overwrites`          | object  |           | Attributes that have organization-level defaults are defined as keys in this object. Each attribute key stores a boolean value that relates to whether or not they are overwriting the default. In order to set one of these attributes to use the inherited default, set it to false. For example, to set `execution-mode` as the organization default: `setting-overwrites.execution-mode = false`. The setting overwrites object is not necessary to include when overwriting the default.                   |

Reviewer 2: I tried shortening this a bit for brevity, I think the points still remain the same but please feel free to push back if you disagree!
```suggestion
| `data.attributes.setting-overwrites`          | object  |           | The keys in this object are attributes that have organization-level defaults. Each attribute key stores a boolean value that relates to whether or not it is overwriting a default. To use the default inherited value, set an attribute's key to false. For example, to set `execution-mode` as the organization default, you set `setting-overwrites.execution-mode = false`. You do not need to include this object when overwriting a default.                   |
```

Reviewer: > The setting overwrites object is not necessary to include when overwriting the default.  

IMO we should omit this information for simplicity. Tell the user if you want to inherit, set the key to false; if you want to overwrite set the key to true.

It happens that they dont need to set the key to true in order to overwrite, for backward compatibility....but I think when we are describing the API it makes the most sense to do so in the simplest way possible.

Alternatively, we could say "The default value for each key is `true`"

What do you think?

Reviewer 2: I like that! I tried rephrasing with this context in mind:
```suggestion
| `data.attributes.setting-overwrites`          | object  |           |  The keys in this object are attributes that have organization-level defaults. Each attribute key stores a boolean value which is `true` by default. To overwrite the default inherited value, set an attribute's value to `false`. For example, to set `execution-mode` as the organization default, you set `setting-overwrites.execution-mode = false`.                       |
```

| `data.attributes.auto-apply`                  | boolean | `false`   | Whether to automatically apply changes when a Terraform plan is successful, [with some exceptions](/terraform/cloud-docs/workspaces/settings#auto-apply-and-manual-apply).                                                                                                                                                                                                                                                                                                                                                |
| `data.attributes.description`                 | string  | (nothing) | A description for the workspace.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                |

> | `data.attributes.execution-mode`              | string  | `remote`  | Which [execution mode](/terraform/cloud-docs/workspaces/settings#execution-mode) to use. Valid values are `default`, `remote`, `local`, and `agent`. When set to `local`, the workspace will be used for state storage only. This value must not be specified if `operations` is specified.                                                                                                                                                                                                                                          |

Reviewer: You can't actually pass `default` directly to the API. I think that we want to leave the valid values the same, and change the `Default` column from `remote` to `(nothing)`. Then we can expand the description to say that this attribute is specifying the workspace level overwrite for the value, and if it is omitted the organization default will be used instead.

}
```

_Using Organization default execution mode_
```json
{
  "data": {
    "attributes": {
      "name":"workspace-with-default",

>       "updated-at": "2023-08-31T19:18:09.976Z",

Reviewer: Bit of a nitpick: we dont need the `udpated-at` here

| `data.attributes.vcs-repo`                    | object or null | (previous value) | To delete a workspace's existing VCS repo, specify `null` instead of an object. To modify a workspace's existing VCS repo, include whichever of the keys below you wish to modify. To add a new VCS repo to a workspace that didn't previously have one, include at least the `oauth-token-id` and `identifier` keys.                                                                                                                                                           |
| `data.attributes.working-directory`           | string         | (previous value) | A relative path that Terraform will execute within. This defaults to the root of your repository and is typically set to a subdirectory matching the environment when multiple environments exist within the same repository.                                                                                                                                                                                                                                                   |

> | `data.attributes.setting-overwrites`          | object  |      | Attributes that have organization-level defaults are defined as keys in this object. Each attribute key stores a boolean value that relates to whether or not they are overwriting the default. In order to set one of these attributes to use the inherited default, set it to false. For example, to set `execution-mode` as the organization default: `setting-overwrites.execution-mode = false`. The setting overwrites object is not necessary to include when overwriting the default. |

Reviewer 2: Matching my above suggestion:
```suggestion
| `data.attributes.setting-overwrites`          | object  |           | The keys in this object are attributes that have organization-level defaults. Each attribute key stores a boolean value that relates to whether or not it is overwriting a default. To use the default inherited value, set an attribute's key to false. For example, to set `execution-mode` as the organization default, you set `setting-overwrites.execution-mode = false`. You do not need to include this object when overwriting a default.                   |
```

Reviewer 2: (This suggestion is updated , make sure it matches whatever we decide on above)

| `data.attributes.auto-apply`                  | boolean        | (previous value) | Whether to automatically apply changes when a Terraform plan is successful, [with some exceptions](/terraform/cloud-docs/workspaces/settings#auto-apply-and-manual-apply).                                                                                                                                                                                                                                                   |
| `data.attributes.description`                 | string         | (previous value) | A description for the workspace.                                                                                                                                                                                                                                                                                                                                                                                   |

> | `data.attributes.execution-mode`              | string         | (previous value) | Which [execution mode](/terraform/cloud-docs/workspaces/settings#execution-mode) to use. Valid values are `default`, remote`, `local`, and `agent`. When set to `local`, the workspace will be used for state storage only. This value must not be specified if `operations` is specified.                                                                                                                                             |

Reviewer: Similar comment here, passing default directly is not allowed

Reviewer: Also, I think we will need to add the `setting-overwrites` attributes to the create/update sections as well.
Just spitballing, but it may be worthwhile adding a subsection about "Workspace Setting Overwrites".

Alternatively (or maybe in addition to that), we could add a new page for "Setting Overwrites" under https://developer.hashicorp.com/terraform/cloud-docs/api-docs (like the overview page), which explains how the setting overwrites work in the API in general

I think we definitely need to tell users "here are the attributes which you can overwrite for the workspace" (at this point just execution mode and agent pool)...how we want to describe the format of the new `setting-overwrites` in general, which is a whole new API concept, is something we could definitely use feedback from @rkoron007 on
